napari_animation/_qt/savedialog_widget.py

```diff
@@ -0,0 +1,98 @@
+from qtpy.QtCore import Qt
+from qtpy.QtWidgets import (
+    QCheckBox,
+    QComboBox,
+    QDoubleSpinBox,
+    QFileDialog,
+    QHBoxLayout,
+    QLabel,
+    QSpinBox,
+    QWidget,
+)
+
+
+class SaveDialogWidget(QFileDialog):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def getSaveFileName(
+        self,
+        parent=None,
+        caption="Select a file :",
+        dir=".",
+        filter="All files (*.*)",
+        selectedFilter="",
+        options=None,
+    ):
+
+        # Set dialog parameters
+        self.setWindowTitle(caption)
+        self.setDirectory(dir)
+        self.setNameFilter(filter)
+        self.setFileMode(QFileDialog.AnyFile)
+        self.setAcceptMode(QFileDialog.AcceptSave)
+
+        if selectedFilter != "":
+            self.selectNameFilter(selectedFilter)
+        if options is not None:
+            self.setOptions(options)
+
+        # add OptionsWidget
+        self.setOption(QFileDialog.DontUseNativeDialog, True)
+        layout = self.layout()
+        self.optionsWidget = OptionsWidget(self)
+        layout.addWidget(self.optionsWidget, 4, 1)
+        self.setLayout(layout)
+
+        # Get info back from user
+        if self.exec_():
+            filename = list(self.selectedFiles())[0]
+            fps = self.optionsWidget.fpsSpinBox.value()
+            quality = int(self.optionsWidget.qualityComboBox.currentText())
+            canvas_only = self.optionsWidget.canvasCheckBox.isChecked()
+            scale_factor = self.optionsWidget.scaleSpinBox.value()
+
+            return filename, fps, quality, canvas_only, scale_factor
```

Comment:
it might be nicer if we return a dict of keyword arguments to `Animation.animate()`

something like

```python
animation_kwargs = {}
animation_kwargs['filename'] = list(self.selectedFiles())[0]
...etc

return animation_kwargs
```
```